jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/IO.java

```diff
@@ -390,6 +394,36 @@ public static boolean delete(File file)
         return file.delete();
     }
 
+    public static boolean delete(Path path)
```

Comment:
Recursive delete of path?   Make sure we don't follow symlinks.

Reply:
This is the equivalent of the pre-existing `boolean delete(File file)` helper. We could skip symlinks but that would make a difference in behavior between the two versions.

Reply:
The old code seems to delete the link, not the destination, as it doesn't use toRealPath or toCanonicalPath or toAbsolutePath in it's logic.

The new code seems to follow the destination, and then delete the destination at the end.
In my eyes, it looks like have a difference in behavior now.
Isn't there an option to NO_FOLLOW_LINKS in an overridden walk method?
and/or is the default behavior to NO_FOLLOW_LINKS with that walk method?
Should we be explicit and declare NO_FOLLOW_LINKS ? (is that even a thing?)
Just nervous I guess.

Reply:
In the Jetty Test Helper, we have a similar recursive delete method.
But it won't delete content that isn't under the start path, it will throw an error if an attempt to delete content somewhere else is detected.

Reply:
I just tried:
```
mkdir -p /tmp/a/b && mkdir -p /tmp/x/y && ln -s /tmp/x /tmp/a/b/x
```
and ran:
```java
IO.delete(Path.of("/tmp/a"));
```
and `/tmp/x` was left untouched. Apparently `Files.walkFileTree` considers symlinks to directories as files and does not follow them by default.